autonat/README.md

```diff
@@ -78,6 +78,12 @@ If all dials fail, the receiver sends a `DialResponse` message with the
 successfully, it sends a `DialResponse` with the `ResponseStatus` `OK`. It
 SHOULD include the address it successfully dialed in its response.
 
+If the receiver elects not to dial any addresses from the message - for example
+because they are invalid, because they resolve to the same host as the receiver,
+because they only include private addresses, because it does not support the
```

Comment:
```suggestion
because they are invalid, because it does not support the
```

> because they only include private addresses

In rust-libp2p we don't reject requests in this case. Directly after starting to listen, a peer won't know any of their external addresses and instead just their listening addresses (which are all private). The server still attempts to dial them, based on the address it observes this peer on, i.e. it replaces the IP of each address in the message with the observed IP. We do this anyway for DDoS protection (see https://github.com/status-im/nim-libp2p/pull/739#discussion_r917551966; @Tanguy suggested that we also add a section on for this, which imo would make sense).
Only if the observed address is private we may reject the peer (depending on whether this was set in our config or not).
> because they resolve to the same host as the receiver

Again, because we replace the IP with the observed one at least in rust-libp2p we don't really handle this case.

---
 Not sure how I think about having an exemplary list of rejection reasons. They seem rather go-lib2p2p specific right now. I'd propose to either properly specify in which cases a DialRequest shall be rejected, or keep it on a higher level to leave it up to the implementation (see suggestion).